src/Header.tsx

```diff
@@ -1,10 +1,15 @@
 import styled, {css} from 'styled-components'
 import PropTypes from 'prop-types'
 import theme from './theme'
-import {get, COMMON, TYPOGRAPHY, BORDER} from './constants'
-import sx from './sx'
+import {BORDER, COMMON, get, SystemCommonProps, SystemTypographyProps, SystemBorderProps, TYPOGRAPHY} from './constants'
+import {ComponentProps} from './utils/types'
+import sx, {SxProp} from './sx'
 
-const Header = styled.div`
+type StyledHeaderItemProps = SystemCommonProps & SxProp & {full?: boolean}
```

Comment:
I like to put the custom props at the beginning of the type definition because I think it's a little easier to read:
```suggestion
type StyledHeaderItemProps = {full?: boolean} & SystemCommonProps & SxProp 
```